x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/bulk_actions/use_bulk_duplicate_confirmation.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useCallback, useRef } from 'react';
+
+import { useBoolState } from '../../../../../common/hooks/use_bool_state';
+
+/**
+ * hook that controls bulk duplicate actions exceptions confirmation modal window and its content
+ */
+export const useBulkDuplicateExceptionsConfirmation = () => {
+  const [isBulkDuplicateConfirmationVisible, showModal, hideModal] = useBoolState();
+  const confirmationPromiseRef = useRef<(result: string | null) => void>();
+
+  const onConfirm = useCallback((value: string) => {
+    confirmationPromiseRef.current?.(value);
+  }, []);
+
+  const onCancel = useCallback(() => {
+    confirmationPromiseRef.current?.(null);
+  }, []);
+
+  const initModal = useCallback(() => {
+    showModal();
+
+    return new Promise<string | null>((resolve) => {
+      confirmationPromiseRef.current = resolve;
```

Comment:
Not sure how this Promise resolves to a string by assigning resolve to the ref :O

It works, so I must not be understanding something here, so maybe a quick explanation?

Reply:
So resolve here will be called later in `onConfirm` and `onCancel` with string or with null value.

It's done on this way, that you can show the modal window, and then wait for confirm or cancel, to get the value

```
const duplicateExceptions = await showBulkDuplicateConfirmation();
        if (!duplicateExceptions) {
          return;
        }
        await executeBulkAction({
          type: BulkActionType.duplicate,
          duplicatePayload: {
            include_exceptions: duplicateExceptions === DUPLICATE_OPTIONS.WITH_EXCEPTIONS,
          },
          ...(isAllSelected ? { query: filterQuery } : { ids: selectedRuleIds }),
        });
```